File: Tree/tree.py

class Binary_Tree:
    def __init__(self, val=0, left=None, right=None) -> None:
        self.val = val
        self.left = left
        self.right = right
    
    
class n_ary_Tree:
    def __init__(self, val=0, child: list=[]) -> None:
        self.val = val
        self.child = child

File: Tree/tree.cpp
#include <vector>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
    ~TreeNode() {delete left, right, val;};
};

struct nary_tree {
    int val;
    vector<nary_tree> child;
    nary_tree() : val(0), child({}) {}
    nary_tree(int x) : val(x), child({}) {}
    nary_tree(int x, vector<nary_tree> &child) : val(x), child(child) {}
    ~nary_tree() {;};
};

